src/runtime.js

```diff
@@ -151,6 +151,7 @@ let tornado = {
    */
   exists(val, placeholderNode, bodies, context) {
     if (this.util.isPromise(val)) {
+      placeholderNode = this.insertPendingBody(placeholderNode, bodies.pending, context) || placeholderNode;
```

Comment:
Alright, I see what you mean. So, should we get rid of `exists` and `section`? I think there is value in making the most common use cases a part of the runtime, allowing them to be optimized beyond what is possible in a helper.